torch/_inductor/codegen/common.py

```diff
@@ -36,6 +44,114 @@ def index_prevent_reordering(index: typing.List[sympy.Expr], index_vars, sizes):
     return [*index, sympy_dot(index_vars, FlexibleLayout.contiguous_strides(sizes))]
 
 
+def _data_type_propagation(sub_graph: torch.fx.Graph):
+    def propagate_node(node: torch.fx.Node):
+        _node: torch.fx.Node = node
+        if _node.target == "ops":
+            return False
+        ops_to_bool = [
+            "is_inf",
+            "is_nan",
+            "bitwise_xor",
+            "logical_not",
+            "signbit",
+            "le",
+            "lt",
+            "ge",
+            "gt",
+            "eq",
+            "ne",
+            "bitwise_not",
+            "bitwise_or",
+            "bitwise_left_shift",
+            "bitwise_right_shift",
+        ]
+        ops_with_dtype_arg = ["constant", "to_dtype", "rand", "randn"]
+        reduction_to_bool = ["any"]
+        reduction_to_int64 = ["argmin", "argmax"]
```

Comment:
Simpler to build a dict from reduction type to dtype?

Reply:
Done